### PR TITLE
Adding gcp queue to filer

### PR DESCRIPTION
**Pull request by thorwolpert**
*Issue #:* n/a

*Description of changes:*
- adding gcp_queue to the filer
- added publish configs
- updated tests
- added test-container
- changed publish_event to publish to the GCP Queue

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: queue_services/entity-filer/src/entity_filer/config.py [hunk 1]
```diff
@@ -131,6 +131,21 @@ class _Config():  # pylint: disable=too-few-public-methods
...
+    BUSINESS_PAY_TOPIC = os.getenv('BUSINESS_PAY_TOPIC', 'business-pay-dev')
+    NAMEX_PAY_TOPIC = os.getenv('NAMEX_PAY_TOPIC', 'namex-pay-dev')
+
+    AUDIENCE = os.getenv(
+        "AUDIENCE", "https://pubsub.googleapis.com/google.pubsub.v1.Subscriber"
+    )
+    PUBLISHER_AUDIENCE = os.getenv(
+        "PUBLISHER_AUDIENCE", "https://pubsub.googleapis.com/google.pubsub.v1.Publisher"
+    )
+    SUB_AUDIENCE = os.getenv("SUB_AUDIENCE", "")
```
Comment (pwei1018) [inline queue_services/entity-filer/src/entity_filer/config.py:146]: vault doesn't have this key.

File: queue_services/entity-filer/src/entity_filer/config.py [hunk 2]
```diff
@@ -131,6 +131,21 @@ class _Config():  # pylint: disable=too-few-public-methods
...
+    SUB_SERVICE_ACCOUNT = os.getenv("SUB_SERVICE_ACCOUNT", "")
```
Comment (pwei1018) [inline queue_services/entity-filer/src/entity_filer/config.py:147]: this key may be call "BUSINESS_SERVICE_ACCOUNT".

File: queue_services/entity-filer/requirements.txt [hunk 3]
```diff
@@ -24,6 +23,7 @@ urllib3==1.26.11
 minio==7.0.2
 PyPDF2==1.26.0
 reportlab==3.6.12
+git+https://github.com/bcgov/sbc-connect-common.git#egg=gcp-queue&subdirectory=python/gcp-queue
 git+https://github.com/bcgov/business-schemas.git@2.18.27#egg=registry_schemas
-git+https://github.com/bcgov/lear.git#egg=legal_api&subdirectory=legal-api
 git+https://github.com/bcgov/lear.git#egg=entity_queue_common&subdirectory=queue_services/common
+git+https://github.com/thorwolpert/lear.git@requirements-split-nats-2#egg=legal_api&subdirectory=legal-api
```
Comment (argush3) [inline queue_services/entity-filer/requirements.txt:29]: can point back to using bcgov LEAR repo after https://github.com/bcgov/lear/pull/2744 is merged.

File: queue_services/entity-filer/src/entity_filer/worker.py [hunk 4]
```diff
@@ -96,23 +102,22 @@ def get_filing_types(legal_filings: dict):
     return filing_types
 
 
-async def publish_event(business: Business, filing: Filing):
+def publish_event(business: Business, filing: Filing):
```
Comment (argush3) [inline queue_services/entity-filer/src/entity_filer/worker.py:105]: I think we have `entity-digital-credentials` listener is listening for nats queue messages.  Maybe `entity-bn` listener too.

Comment (thorwolpert) [reply queue_services/entity-filer/src/entity_filer/worker.py:105]: Are they listening on the generic queue?
I though the plan was for those to had dedicated topics, or is that a future?

Comment (argush3) [reply queue_services/entity-filer/src/entity_filer/worker.py:105]: think there both listening on generic events queue.  

@vysakh-menon-aot think you are more familiar with entity-bn

Comment (vysakh-menon-aot) [reply queue_services/entity-filer/src/entity_filer/worker.py:105]: yes in entity-bn we are depending on it

Comment (argush3) [reply queue_services/entity-filer/src/entity_filer/worker.py:105]: Thanks for checking Vysakh!

Thor - as per our conversation, let's add back the NATS message so both(GCP & NATS) messages are sent for each event.  Thanks!

Comment (vysakh-menon-aot) [reply queue_services/entity-filer/src/entity_filer/worker.py:105]: example https://github.com/bcgov/lear/blob/main/legal-api/src/legal_api/resources/v2/admin/administrative_bn.py#L44

**Review by argush3 — APPROVED** [review]

**Review by vysakh-menon-aot — APPROVED** [review]

**Pull request merged**